### PR TITLE
Feature/webops 322 demote site envs

### PR DESCRIPTION
**Pull request by pingers**
To test:
* Run robo dev:content-generate
* Create another environment for the default site.
* Create another separate test site and environment.
* Promote dev environments from both sites.
* Promote dev environment from a site with another prod environment.

Pass:
* Each site has at most 1 prod environment.
* Site A prod environment does not get demoted when Site B dev environment gets promoted.
* Site A prod environment gets demoted when Site A dev environment gets promoted.

Fail:
* Site A dev environment causes ALL other prod environments to get demoted to dev.

### CODE REVIEW

File: web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php [hunk 1]
```diff
@@ -664,7 +664,7 @@ public function getEnvironmentStatus(string $project_name, string $short_name, s
       return FALSE;
     }
 
-    return $this->extractDeploymentConfigStatus($deployment_config);
+    return $deployment_config ? $this->extractDeploymentConfigStatus($deployment_config) : FALSE;
```
Comment (pingers) [inline web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: Sorry, this was annoying me. Had to fix it. extractDeploymentConfigStatus() expects array and was being passed boolean when there was a problem creating the deployment config in openshift causing WSOD on sites/{nid}/environments.

Comment (singularo) [reply web/modules/custom/shepherd/shp_orchestration/src/Plugin/OrchestrationProvider/OpenShiftOrchestrationProvider.php]: hehe

File: web/modules/custom/shepherd/shp_orchestration/src/Service/Environment.php [hunk 2]
```diff
@@ -306,8 +306,9 @@ public function promoted(NodeInterface $site, NodeInterface $environment, bool $
       ->execute();
     $promoted_term = reset(Term::loadMultiple($ids));
 
-    // Demote all current prod environments.
+    // Demote all current prod environments - for this site!
```
Comment (singularo) [inline web/modules/custom/shepherd/shp_orchestration/src/Service/Environment.php]: Ahah. yeah, the problem with testing on a small (1) number of sites.

**Comment by singularo** [comment]
Fixes the promote/demote issue.
:shipit:

**Comment by singularo** [comment]
It might be a 3.9 api 'feature', but it looks like deleting an environment doesn't removes it from openshift anymore ;-(

**Comment by pingers** [comment]
That's a 3.7 issue and yeah, reported in another issue ;) Separate MR.
It's just the replication controller part I believe.

**Pull request merged**